staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -241,10 +250,18 @@ func (s *Validator) validateMap(ctx context.Context, fldPath *field.Path, sts *s
 		return nil, remainingBudget
 	}
 
+	// if a third map type is introduced, assume it's not correlatable. granular is the default if unspecified.
+	correlatable := sts.XMapType == nil || *sts.XMapType == "granular" || *sts.XMapType == "atomic"
```

Comment:
One is for Structural the other is for JSONSchemaProps.  Do we need these guards? Adding an additional map type would break the world far beyond what this code guards.